core/services/keystore/eth_v2.go

```diff
@@ -0,0 +1,359 @@
+package keystore
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ethkey"
+	"gorm.io/gorm"
+)
+
+//go:generate mockery --name Eth --output mocks/ --case=underscore
+
+// Eth is the external interface for EthKeyStore
+type Eth interface {
+	Get(id string) (ethkey.KeyV2, error)
+	GetAll() ([]ethkey.KeyV2, error)
+	Create() (ethkey.KeyV2, error)
+	Add(key ethkey.KeyV2) error
+	Delete(id string) (ethkey.KeyV2, error)
+	Import(keyJSON []byte, password string) (ethkey.KeyV2, error)
+	Export(id string, password string) ([]byte, error)
+
+	EnsureKeys() (ethkey.KeyV2, bool, ethkey.KeyV2, bool, error)
+	SubscribeToKeyChanges() (ch chan struct{}, unsub func())
+
+	SignTx(fromAddress common.Address, tx *types.Transaction, chainID *big.Int) (*types.Transaction, error)
+
+	SendingKeys() (keys []ethkey.KeyV2, err error)
+	FundingKeys() (keys []ethkey.KeyV2, err error)
+	HasSendingKeyWithAddress(address common.Address) (bool, error)
+	GetRoundRobinAddress(addresses ...common.Address) (address common.Address, err error)
+
+	GetState(id string) (ethkey.State, error)
+	GetStatesForKeys([]ethkey.KeyV2) ([]ethkey.State, error)
```

Comment:
Can we have these return ID (and maybe even drop the ethkey.State object entirely) because this falsely implies the state might be up-to-date/

Reply:
Ahh I see now.. let me look into where all this is used and see if we that makes more sense

Reply:
Sorry I wasn't clear. I don't mean drop the table - we need that. And I think `state` is the correct name.

I meant, perhaps we don't need the struct - since state updates ought to hit the database directly to get the most up-to-date values

Reply:
LastUsed is no longer persisted to the DB